### ISSUE TITLE
.find('> bar') selects also nested descendants

### ISSUE THREAD

**Issue by algesten** [cheeriojs/cheerio#742]
In this example I expect `> bar` to only select the *immediate descendant* from `foo`. But I get also the nested descendant. Cheerio 0.19.0

```xml
<foo>
    <bar>
        <bar></bar>
    </bar>
</foo>
```

```coffee
$ = require('cheerio').load('<foo><bar><bar></bar></bar></foo>')
$('foo').find('> bar').length
2   # I expect 1
```

**Comment by fb55** [comment]
This is already fixed in the `css-select` repo, but triggered another bug that has to be fixed before the feature is available in cheerio.

**Comment by clkao** [comment]
a release with this fix seems to depend on #728

**Comment by algesten** [comment]
I wrote my own jquery-on-the-server partly because I got frustrated by this. https://github.com/algesten/zu
